File: C++/1310. XOR Queries of a Subarray.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& a, vector<vector<int>>& q) {
        vector<int>v;       
      int z=0;
        for(int i=0;i<a.size();i++)
        {
            z=z^a[i];
            v.push_back(z);
        }
                vector<int>ans;        
        for(int i=0;i<q.size();i++)
        {
            int a=q[i][0];
            int b=q[i][1];
            int k=0;
            if(a-1>=0)
             k=v[b]^v[a-1];
            else
                k=v[b];
            ans.push_back(k);
        }
        return ans;
    }
};

File: JAVA/1310. XOR Queries of a Subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int n = arr.length;
        int[] prefix = new int[n];
        prefix[0] = arr[0];
        for (int i = 1; i < n; i++) {
            prefix[i] = prefix[i - 1] ^ arr[i];
        }
        int[] result = new int[queries.length];        
        for (int i = 0; i < queries.length; i++) {
            int left = queries[i][0];
            int right = queries[i][1];           
            if (left == 0) {
                result[i] = prefix[right];
            } else {
                result[i] = prefix[right] ^ prefix[left - 1];
            }
        }        
        return result;
    }
}
